library_generation/model/generation_config.py

```diff
@@ -130,7 +142,9 @@ def from_yaml(path_to_yaml: str) -> GenerationConfig:
 
 def __required(config: Dict, key: str):
     if key not in config:
-        raise ValueError(f"required key {key} not found in yaml")
+        raise ValueError(
+            f"required key {key} not found in {config} " f"when parsing yaml"
```

Comment:
Is {config} going to print out the whole config? If yes, I don't think we want to do that. Something like 
```
f"Required repo level config {key} not found"
```
should be good enough.

Reply:
Separately, there are a few library level required field, and I don't see any validation for them yet. They can be addressed in a separate PR.

Reply:
I remove `config` in the err message.

> Separately, there are a few library level required field, and I don't see any validation for them yet. They can be addressed in a separate PR.

Library level required fields are tests in `utilities_unit_tests.py`. I moved it to `generation_config_unit_test.py`.

Reply:
I see that library level config is validated in the same way as repo level config now. However, it makes the error message not clear for library level config. e.g. If `api-shortname` is missing for a specific library, we don't know which library is missing `api-shortname` from `required key {key} not found`.